### DISCUSSION TITLE
app::set_focus() for Group?

### DISCUSSION THREAD

**Discussion by wyhinton** [fltk-rs/fltk-rs#762] [Q&A]
What's the proper way to use ```app::set_focus()``` on a ```&mut Group```/```&mut Pack```? Do we have to upcast it to Widget?

Also are  ```app::set_focus(&widg)``` and ```widg.take_focus()``` achieving the same thing?
```rust
use fltk::{
    app, app::*, button::*, enums::*, frame::*, group::*, prelude::*, window::*, widget::*
};

#[derive(Clone)]
pub enum Message {
    SetDragTarget(String),
}
fn main(){
    let app = App::default();
    let mut win = Window::new(200, 200, 500, 300, "Drag and Move ");

    let mut group = Group::new(100,100,100,100, "MyGroup");
    group.set_frame(FrameType::FlatBox);
    group.set_color(Color::Blue);
    group.set_align(Align::Inside|Align::Center);
    group.handle(move|widg, ev|{
      match ev{
        Event::Enter=>{
            widg.set_color(Color::Red);
            widg.redraw();
            let w = unsafe {Widget::from_widget_ptr(widg.as_widget_ptr()) };
            app::set_focus(&w);
        true
        }
        _=>false              
      }
    });

    group.end();

    win.end();
    win.show();

    app.run().unwrap();
}
```

**Comment by MoAlyousef** [comment]
Normally you can use app::set_focus(widg) directly (no need for the ampersand since widg is a reference itself. 
app::set_focus is meant for application startup, setting the focus on a specific widget, otherwise you would use take_focus(). Some widgets don’t accept focus, not sure if group widgets do. I can’t test at the moment. But you can test with take_focus() since it returns a Result.
